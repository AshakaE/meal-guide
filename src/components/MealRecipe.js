import React from 'react';
import { useParams } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getRecipe } from '../actions/index';
import Loading from './Loading';
import RecipeCard from './RecipeCard';
import vals from '../assets/values';

const MealRecipe = (props) => {
  const { getRecipe, loading, recipe } = props;
  const { id } = useParams();
  const cas = vals(recipe);
  console.log(cas[0]);
  React.useEffect(() => {
    getRecipe(`${id}`);
  }, [getRecipe]);
  if (loading) {
    return <Loading />;
  }
  return (
    <div>
      {Object.values(cas).map((recipe) => (
        <RecipeCard key={recipe.id} recipe={recipe} />
      ))}
    </div>
  );
};

const mapStateToProps = ({ recipeState: { recipe, loading } }) => ({
  loading,
  recipe,
});

MealRecipe.propTypes = {
  loading: PropTypes.bool.isRequired,
  recipe: PropTypes.arrayOf(PropTypes.object).isRequired,
  getRecipe: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, { getRecipe })(MealRecipe);
