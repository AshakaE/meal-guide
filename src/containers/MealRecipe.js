import React from 'react';
import { useParams } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getRecipe } from '../actions/index';
import Loading from '../components/Loading';
import RecipeCard from '../components/RecipeCard';
import vals from '../assets/values';

const MealRecipe = (props) => {
  const { getRecipe, loading, recipe } = props;
  const { id } = useParams();
  const cas = vals(recipe);

  React.useEffect(() => {
    getRecipe(`${id}`);
  }, [getRecipe]);
  if (loading) {
    return <Loading />;
  }
  return (
    <div>
      {Object.values(cas).map((recipe) => (
        <RecipeCard key={recipe.id} recipe={recipe} />
      ))}
    </div>
  );
};

const mapStateToProps = ({ recipeState: { recipe, loading } }) => ({
  loading,
  recipe,
});

MealRecipe.propTypes = {
  loading: PropTypes.bool.isRequired,
  recipe: PropTypes.arrayOf(PropTypes.object).isRequired,
  getRecipe: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, { getRecipe })(MealRecipe);
