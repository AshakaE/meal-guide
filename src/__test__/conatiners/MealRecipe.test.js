import React from 'react';
import renderer from 'react-test-renderer';
import { createStore, combineReducers, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import ReactRouter from 'react-router';
import mealReducer from '../../reducers/meal';
import recipeReducer from '../../reducers/recipe';
import filterReducer from '../../reducers/filter';
import MealRecipe from '../../components/MealRecipe';

const middleware = [thunk];
const store = createStore(
  combineReducers({
    mealState: mealReducer,
    recipeState: recipeReducer,
    filterState: filterReducer,
  }),
  composeWithDevTools(applyMiddleware(...middleware)),
);

describe('MealRecipe', () => {
  it('renders correctly', () => {
    jest.spyOn(ReactRouter, 'useParams').mockReturnValue({ id: '1234' });
    const tree = renderer
      .create(
        <MealRecipe
          store={store}
          id="1234"
          title="Test meal"
          image="url"
          cuisine="cuisine"
          dish={['dish']}
          meal={['meal']}
          key="1"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
