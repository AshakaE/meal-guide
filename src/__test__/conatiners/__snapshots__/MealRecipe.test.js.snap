// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealRecipe renders correctly 1`] = `<div />`;
